src/main/java/com/robertx22/mine_and_slash/a_libraries/jei/RunewordCategory.java

```diff
@@ -0,0 +1,108 @@
+package com.robertx22.mine_and_slash.a_libraries.jei;
+
+import com.robertx22.mine_and_slash.database.data.runewords.RunewordRecipe;
+import com.robertx22.mine_and_slash.database.registry.ExileDB;
+import com.robertx22.mine_and_slash.mmorpg.registers.common.items.RuneItems;
+import com.robertx22.mine_and_slash.uncommon.localization.Words;
+import mezz.jei.api.gui.builder.IRecipeLayoutBuilder;
+import mezz.jei.api.gui.drawable.IDrawable;
+import mezz.jei.api.gui.ingredient.IRecipeSlotsView;
+import mezz.jei.api.helpers.IGuiHelper;
+import mezz.jei.api.recipe.IFocusGroup;
+import mezz.jei.api.recipe.RecipeType;
+import mezz.jei.api.recipe.category.IRecipeCategory;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class RunewordCategory implements IRecipeCategory<RunewordRecipe> {
+    final IDrawable background;
+    IGuiHelper guiHelper;
+    RecipeType<RunewordRecipe> recipeType;
+    final IDrawable icon;
+
+    RunewordExtension recipeExtension;
+
+    RunewordCategory(IGuiHelper guiHelper, RecipeType<RunewordRecipe> type) {
+        this.guiHelper = guiHelper;
+        this.recipeType = type;
+        icon = guiHelper.createDrawableItemStack(RuneItems.MAP.entrySet().stream().findFirst().get().getValue().get().getDefaultInstance());
+
+        Dimensions dimension = calculateDimension();
+        int maxWidth = dimension.width;
+        int maxLines = dimension.lines;
+
+        recipeExtension = new RunewordExtension(maxWidth);
+        var height = recipeExtension.calculateMaxHeight(maxLines);
+        background = guiHelper.createBlankDrawable(maxWidth, height);
+    }
+
+    private Dimensions calculateDimension() {
+        AtomicInteger maxWidth = new AtomicInteger();
+        AtomicInteger maxLines = new AtomicInteger();
+        var font = Minecraft.getInstance().font;
+        ExileDB.RuneWords().getList().forEach(word -> {
+            word.stats.forEach(stat -> {
```

Comment:
Yea, lol, but need this to add "find usages" feature for all items that can be a host for runeword.
No idea if we need to cache this, what's more important processing speed or memory?